acc_test/resource_kafka_user_test.go

```diff
@@ -77,6 +85,10 @@ func TestKafkaUserResource(t *testing.T) {
 				Config: updateKafkaUserConfig,
 				Check:  checkKafkaUserUpdated(newPassword),
 			},
+			{
+				Config: invalidKafkaUserCreateConfigDuplicate,
+				ExpectError: regexp.MustCompile("A Kafka user with this username already exists on this cluster."),
```

Comment:
question - is this a desired format of error message? in this situation, terraform provider will just return the API errer response, however it seems like it used to be handled on the provider side as well in the code below

```go
// just use linear search for now to check if the user going to be created is already in the system
	usernameList, err := client.ReadKafkaUserList(cluster_id)
	if err != nil {
		return fmt.Errorf("[Error] Error retrieving kafka user list: %s", err)
	}
	for _, str := range usernameList {
		if str == username {
			// user is already set, so we don't change anything
			d.SetId(fmt.Sprintf("%s-%s", cluster_id, username))
			log.Printf("[INFO] Kafka user %s already exists in %s.", username, cluster_id)
			return nil
		}
	}
```

wondering if that should be modified to fail with graceful error message crafted in the terraform provider instead of removing it?

Reply:
So we removed the handling in terraform provider in favour of the handling in the API side. The API returns a proper error instead of 500, so the error should propagate